src/ert/shared/storage/key_utils.py

```diff
@@ -0,0 +1,199 @@
+from enum import Enum, auto
+from typing import List
+
+special_vars = [
+    "NAIMFRAC",
+    "NBAKFL",
+    "NBYTOT",
+    "NCPRLINS",
+    "NEWTFL",
+    "NEWTON",
+    "NLINEARP",
+    "NLINEARS",
+    "NLINSMAX",
+    "NLINSMIN",
+    "NLRESMAX",
+    "NLRESSUM",
+    "NMESSAGE",
+    "NNUMFL",
+    "NNUMST",
+    "NTS",
+    "NTSECL",
+    "NTSMCL",
+    "NTSPCL",
+    "ELAPSED",
+    "MAXDPR",
+    "MAXDSO",
+    "MAXDSG",
+    "MAXDSW",
+    "STEPTYPE",
+    "WNEWTON",
+]
+rate_vars = [
+    "OPR",
+    "OIR",
+    "OVPR",
+    "OVIR",
+    "OFR",
+    "OPP",
+    "OPI",
+    "OMR",
+    "GPR",
+    "GIR",
+    "GVPR",
+    "GVIR",
+    "GFR",
+    "GPP",
+    "GPI",
+    "GMR",
+    "WGPR",
+    "WGIR",
+    "WPR",
+    "WIR",
+    "WVPR",
+    "WVIR",
+    "WFR",
+    "WPP",
+    "WPI",
+    "WMR",
+    "LPR",
+    "LFR",
+    "VPR",
+    "VIR",
+    "VFR",
+    "GLIR",
+    "RGR",
+    "EGR",
+    "EXGR",
+    "SGR",
+    "GSR",
+    "FGR",
+    "GIMR",
+    "GCR",
+    "NPR",
+    "NIR",
+    "CPR",
+    "CIR",
+    "SIR",
+    "SPR",
+    "TIR",
+    "TPR",
+    "GOR",
+    "WCT",
+    "OGR",
+    "WGR",
+    "GLR",
+]
+
+seg_rate_vars = [
+    "OFR",
+    "GFR",
+    "WFR",
+    "CFR",
+    "SFR",
+    "TFR",
+    "CVPR",
+    "WCT",
+    "GOR",
+    "OGR",
+    "WGR",
+]
+
+
+class SummaryVarType(Enum):
+    RD_SMSPEC_INVALID_VAR = auto()
```

Comment:
RD is a prefix for "RESDATA". We could maybe drop that. Maybe SMSPEC can also be made redundant (it is short for "summary specification")

Reply:
is lower case more pythonic?

Reply:
and VAR postfix should be redundant.

Reply:
I thought about `key`, just wanted to maintain the resdata name convention. But will change to key :+1: